### PR TITLE
spago: init at 0.8.5.0

### PR DESCRIPTION
**Pull request by Shou**
Credit to [justinwoo/easy-purescript-nix][1].

[1]: https://github.com/justinwoo/easy-purescript-nix

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/web/spago/default.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
+{ pkgs ? import <nixpkgs> {} }:
```
Comment (alexarice) [inline pkgs/development/web/spago/default.nix]: You should not import nixpkgs from within nixpkgs

File: pkgs/development/web/spago/default.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  patchelf = libPath: if pkgs.stdenv.isDarwin
+    then ""
+    else ''
+          chmod u+w $SPAGO
+          patchelf --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" --set-rpath ${libPath} $SPAGO
+          chmod u-w $SPAGO
+        '';
+
+in pkgs.stdenv.mkDerivation rec {
+  name = "spago";
```
Comment (alexarice) [inline pkgs/development/web/spago/default.nix]: this should be `pname` instead of `name`

File: pkgs/development/web/spago/default.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    then pkgs.fetchurl {
+      url = "https://github.com/spacchetti/spago/releases/download/0.8.5.0/osx.tar.gz";
+      sha256 = "0a6s4xpzdvbyh16ffcn0qsc3f9q15chg0qfaxhrgc8a7qg84ym5n";
+    }
+    else pkgs.fetchurl {
+      url = "https://github.com/spacchetti/spago/releases/download/0.8.5.0/linux.tar.gz";
+      sha256 = "0r66pmjfwv89c1h71s95nkq9hgbk7b8h9sk05bfmhsx2gprnd3bq";
+    };
+
+  buildInputs = [ pkgs.gmp pkgs.zlib pkgs.ncurses5 pkgs.stdenv.cc.cc.lib ];
```
Comment (alexarice) [inline pkgs/development/web/spago/default.nix]: unsure about the c compiler library package, perhaps this should be a native build input?

**Review by alexarice — COMMENTED** [review]
The github page suggests installing this through npm. Would the existing node tools be a better way of installing this

File: pkgs/development/web/spago/default.nix (hunk 3 above)
Comment (Shou) [reply pkgs/development/web/spago/default.nix]: I've done this now.

File: pkgs/development/web/spago/default.nix (hunk 1 above)
Comment (Shou) [reply pkgs/development/web/spago/default.nix]: Yeah, I should've noticed earlier. It's fixed now, thanks!

**Comment by cdepillabout** [comment]
I packaged spago in https://github.com/NixOS/nixpkgs/pull/74557.

However, this builds it from source and doesn't use the easy-purescript-nix approach.

If spago built from source breaks frequently, it might make sense to go back and use the easy-purescript-nix approach.

**Comment by Shou** [comment]
(I'm late but I'll close this now given the above)

**Pull request closed without merging**